services/httpd/response_writer.go

```diff
@@ -131,15 +132,31 @@ func (f *jsonFormatter) WriteResponse(w io.Writer, resp Response) (err error) {
 	}
 
 	if err != nil {
-		_, err = io.WriteString(w, err.Error())
-	} else {
-		_, err = w.Write(b)
+		unnestedErr := unnestError(err)
+		// ignore any errors in this section, we already have a 'real' error to return
+		resp := Response{Err: unnestedErr}
+		if f.Pretty {
+			b, _ = json.MarshalIndent(resp, "", "    ")
+		} else {
+			b, _ = json.Marshal(resp)
+		}
+		w.Write(b)
+		w.Write([]byte("\n"))
+		return err
 	}
 
+	_, err = w.Write(b)
```

Comment:
Now wondering if we should check for an error on the w.Write ([]byte("\n") call.  Or maybe append the \n and only have one write (in both places it is done).

Reply:
Fixed the error check